### PR TITLE
Chapter 24: HTTP and Background Tasks

### PR DESCRIPTION
**Pull request by OnicaAdrianCatalin**
## New features

PhotoGallery is an application that in the end will display images from Flickr API.

In this update we'll just display the title of the images.

## Changes in code

- Added recycler view to display the titles of the images.

- Made the HTTP request to Flickr API.

- Organized the App using Repository pattern.

Screenshot 2022-06-10 at 16 07 42

### CODE REVIEW

**Review by ghost — APPROVED** [review]
Looks good

File: app/build.gradle [hunk 1]
```diff
@@ -32,12 +35,19 @@ android {
 }
 
 dependencies {
-
-    implementation 'androidx.core:core-ktx:1.7.0'
+    implementation 'androidx.core:core-ktx:1.8.0'
```
Comment (NegreaVlad) [inline app/build.gradle:38]: Nice separation of dependencies :)

File: app/src/main/java/com/example/photogallery/presentation/photoGallery/PhotoAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import com.example.photogallery.data.model.GalleryItem
+
+class PhotoAdapter(private val galleryItem: List<GalleryItem>) :
+    RecyclerView.Adapter<PhotoAdapter.PhotoHolder>() {
+
+    class PhotoHolder(itemTextView: TextView) : RecyclerView.ViewHolder(itemTextView) {
+
+        val bindTitle: (String) -> Unit = itemTextView::setText
```
Comment (NegreaVlad) [inline app/src/main/java/com/example/photogallery/presentation/photoGallery/PhotoAdapter.kt:13]: Nice!🚀

**Review by NegreaVlad — APPROVED** [review]
Very nice! Good separation of concerns between view model and fragment.

**Pull request merged**